src/App.tsx

```diff
@@ -1,24 +1,349 @@
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Filters } from './types/Filters';
 import { UserWarning } from './UserWarning';
+import { ErrorMessage } from './types/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 11689;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState(Filters.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [pageIsLoaded, setPageIsLoaded] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isUpdatingId, setIsUpdatingId] = useState<number[]>([]);
+  const [isEditing, setIsEditing] = useState<number | null>(null);
+  const [editQuery, setEditQuery] = useState('');
+  const [editQueryPrev, setEditQueryPrev] = useState('');
+
+  const handleErrorSet = (errMessage: string) => {
+    setErrorMessage(errMessage);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  const closeError = () => {
+    setErrorMessage('');
+  };
+
+  const loadTodos = useCallback(async () => {
+    try {
+      const loadedTodos = await getTodos(USER_ID);
+
+      setTodos(loadedTodos);
+      setPageIsLoaded(true);
+    } catch {
+      handleErrorSet(ErrorMessage.loadTodo);
+    }
+  }, []);
+
+  useEffect(() => {
+    loadTodos();
+  }, [loadTodos]);
+
+  const visibleTodos = (filterBy: Filters) => {
+    if (todos) {
+      switch (filterBy) {
+        case Filters.All:
+          return todos;
+        case Filters.Active:
+          return todos.filter(todo => (
+            !todo.completed
+          ));
+        case Filters.Completed:
+          return todos.filter(todo => (
+            todo.completed
+          ));
+        default: return todos;
+      }
+    }
+
+    return todos;
+  };
+
+  const handleFormSubmit = async (e?: Event) => {
+    e?.preventDefault();
+    if (!query.trim().length) {
+      handleErrorSet(ErrorMessage.titleEmpty);
+
+      return;
+    }
+
+    try {
+      const newTodo = {
+        userId: USER_ID,
+        title: query.trim(),
+        completed: false,
+      };
+
+      setTempTodo({ ...newTodo, id: 0 });
+      setPageIsLoaded(false);
+
+      const newTodos = await addTodo('/todos', newTodo);
+
+      setTodos(state => {
+        return [...state, newTodos as Todo];
+      });
+
+      setTempTodo(null);
+      setPageIsLoaded(true);
+      setQuery('');
+    } catch {
+      handleErrorSet(ErrorMessage.addTodo);
+      setTempTodo(null);
+      setPageIsLoaded(true);
+    }
+  };
+
+  const handleComplete = async (id: number, completed: boolean) => {
+    try {
+      setIsUpdatingId(state => (
+        [...state, id]
+      ));
+      await updateTodo(id, { completed: !completed });
+      setTodos(state => {
+        state.map(todo => {
+          if (todo.id === id) {
+            // eslint-disable-next-line no-param-reassign
+            todo.completed = !completed;
```

Comment:
Don't ignore eslint rule, try to fix them
```suggestion
            return {
              ...todo,
              completed: !completed,
            }
```